### ISSUE TITLE
java.lang.IllegalStateException: Method addObserver must be called on the main thread with SDK version > 2.3.0

### ISSUE THREAD

**Issue by BinhTNT** [helpscout/beacon-android-sdk-sample#162]
Hello,
 
I have encountered an crash issue related to Location permission when using SDK latest version. It's blocking us to upgrade latest SDK to resolve many fixed issues. The issue does not happen if I use SDK version < 2.3.0. I have no clue on what is the relationship between Location permission and latest SDK. Could you please help ?
 
- Beacon version: 2.3.2 or 2.3.3

**Actual outcome**

    The app crash when asking location permission

**Expected outcome**

    The app should not crash when asking location permission

**Steps to reproduce**

    1. Make sure the app is using latest SDK version (2.3.2 or 2.3.3) and make sure the Location is off on device
    2. By somehow make the app asking Location permission when running the app
    3. Notice that app crash with exception "Method addObserver must be called on the main thread"

**Screenshots/Screen Recording**

    
![Screenshot_20210305-115719](https://user-images.githubusercontent.com/45462932/110277375-b6818500-8007-11eb-9726-78a359311d43.png)

**Additional Information**

java.lang.IllegalStateException: Method addObserver must be called on the main thread
    at androidx.lifecycle.LifecycleRegistry.enforceMainThreadIfNeeded(LifecycleRegistry.java:317)
    at androidx.lifecycle.LifecycleRegistry.addObserver(LifecycleRegistry.java:172)
    at androidx.fragment.app.Fragment.initLifecycle(Fragment.java:496)
    at androidx.fragment.app.Fragment.<init>(Fragment.java:476)
    at com.google.android.gms.common.api.internal.zzc.<init>(com.google.android.gms:play-services-basement@@17.2.1:1)
    at com.google.android.gms.common.api.internal.zzc.zza(com.google.android.gms:play-services-basement@@17.2.1:14)
    at com.google.android.gms.common.api.internal.LifecycleCallback.getFragment(com.google.android.gms:play-services-basement@@17.2.1:3)
    at com.google.android.gms.common.api.internal.LifecycleCallback.getFragment(com.google.android.gms:play-services-basement@@17.2.1:7)
    at com.google.android.gms.common.api.internal.zaae.zaa
    at com.google.android.gms.common.api.GoogleApi.<init>
    at com.google.android.gms.common.api.GoogleApi.<init>
    at com.google.android.gms.location.SettingsClient.<init>
    at com.google.android.gms.location.LocationServices.getSettingsClient
    at com.heanoria.library.reactnative.locationenabler.RNAndroidLocationEnablerModule.promptForEnableLocationIfNeeded(RNAndroidLocationEnablerModule.java:73)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
    at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:151)
    at com.facebook.react.bridge.queue.NativeRunnable.run(NativeRunnable.java)
    at android.os.Handler.handleCallback(Handler.java:751)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
    at android.os.Looper.loop(Looper.java:154)
    at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:226)
    at java.lang.Thread.run(Thread.java:761)
    
    
Thanks,
Binh

**Comment by shaunmcilroy** [comment]
Hi @BinhTNT 

We've chatted about this report internally this morning and can confirm there is no relationship between the Beacon SDK and the Location permission.

It's possible this issue is related our SDK using a more up-to-date version of Firebase Messaging (v21.0.1) and this could be pulling in some other dependancies you're also using.

Our next step is to recommend you try updating to the latest versions of Firebase/Google Play services, as Beacon SDK is using 21.0.1 as noted above. Our thinking here is that when building locally Gradle may be resolving the newer version which potentially might not be compatible with your current setup.

Alternatively, if that's not resolving the issue can you fork the sample repo/apps and add your location request code to see if it's a reproducible issue? If that's something you can verify we'll then take a look.

**Comment by BinhTNT** [comment]
Hi @shaunmcilroy Thanks for your comment. I found that our project has a 3rd lib (we are using React Native) which is using "com.google.android.gms:play-services-location" dependencies with version old version and I think it's not compatible with latest SDK. I tried to increase it to 18.0.0 and luckily the crash was fixed! Sorry for posting an issue that is not really to Beacon SDK, but at least thanks to what you mentioned, I knew that I need to take a look on Google Play services dependencies. Thanks a lot for your help!
